dlib/gui_widgets/widgets.cpp

```diff
@@ -6059,6 +6062,15 @@ namespace dlib
         overlay_rects[selected_rect].parts[part_name] = c1;
         parent.invalidate_rectangle(rect); 
 
+        // trz
+        // I do this here so that it advances when parts
+        // are added from the context menu too
+        // This also allows to restart the sequence from
+        // a random point using the context menu
+        int part_idx = std::distance(part_names.begin(), part_names.find(part_name));
+        //printf("part_idx: %d\n", part_idx);
```

Comment:
Don't leave commented out code

Reply:
I've added the reset in the rectangle selection code. Are there any other cases? I couldn't find any after looking at the code and playing with the program.